src/bindings/fluid_cmd.c

```diff
@@ -59,6 +59,17 @@ struct _fluid_shell_t {
   fluid_ostream_t out;
 };
 
+/**
+ * Reduced command information structure for constant data.
+ * For internal use only.
+ */
+typedef struct {
+  const char *name;             /**< The name of the command, as typed in the shell */
+  const char *topic;            /**< The help topic group of this command */ 
+  fluid_cmd_func_t handler;     /**< Pointer to the handler for this command */
+  const char *help;             /**< A help string */
+} fluid_cmd_int_t;
```

Comment:
You want to optimize in size, that's ok. But I dont like that code duplication. I would rather prefer to somehow have the public struct re-using the internal struct. However anonymous structs are a non-standard extension. So I currently cant offer a really better alternative, expect than some fancy preprocessor-macro-hack.

I'm even thinking about removing `fluid_cmd_handler_register()` from the public API. Cant think of any use-case nor of any downstream-application using it.

Reply:
In my opinion, it looks a code duplication, but just apparently.
Besides the deletion of the 'data' field, at the end the real purpose is the declaration of a container for const objects and this have to be necessarily done in this way, while the public fluid_cmd_t has not this strict requirement, at least this is what I have understood after reading the sources, so the global structure cannot be used for this purpose. Actually, I do not know many other ways to do that.

Reply:
Ok, just leave it for now. Will perhaps address that later.